platform/src/main/python/dlpx/virtualization/platform/_linked.py

```diff
@@ -237,6 +248,57 @@ def to_protobuf(snapshot):
 
         return direct_post_snapshot_response
 
+    def _internal_direct_source_size(self, request):
+        """Direct Source Size Wrapper for direct plugins.
+
+        Executed as part of several operations to get the source size
+        of a direct source
+
+        Run source_size operation for a direct source.
+
+        Args:
+           request (DirectSourceSizeRequest): Source Size Request arguments.
+
+        Returns:
+           DirectSourceSizeResponse: A response containing the return value -
+           DirectSourceSizeResult which has source size. In
+           case of errors, response object will contain PluginErrorResult.
+        """
+        # Reasoning for method imports are in this file's docstring.
+        from generated.definitions import RepositoryDefinition
+        from generated.definitions import LinkedSourceDefinition
+        from generated.definitions import SourceConfigDefinition
+
+        #
+        # While linked.source_size() is not a required operation, this should
+        # not be called if it wasn't implemented.
+        #
+        if not self.source_size_impl:
+            raise OperationNotDefinedError(Op.LINKED_SOURCE_SIZE)
+
+        direct_source_definition = LinkedSourceDefinition.from_dict(
+            json.loads(request.direct_source.linked_source.parameters.json))
+        direct_source = DirectSource(
+            guid=request.direct_source.linked_source.guid,
+            connection=RemoteConnection.from_proto(
+                request.direct_source.connection),
+            parameters=direct_source_definition)
+
+        repository = RepositoryDefinition.from_dict(
+            json.loads(request.repository.parameters.json))
+        source_config = SourceConfigDefinition.from_dict(
+            json.loads(request.source_config.parameters.json))
+
+        size = self.source_size_impl(
```

Comment:
nit: change "size" to "source_size"

Reply:
Sure, will be taking care in separate JIRA.